Subject: [SLERT] Patch down RT priority to avoid high-priority spinner on SLERT 
CC: Stefan Assmann <sassmann@suse.de>
CC: slert-devel@suse.de

TODO_SIGNOFF
REVIEW_ON_SLERT: This should got fixed with removing affintiy of IRQ threads in SLERT 10 SP2 -> Obsolete patch?

> Tue Oct  9 15:23:47 CEST 2007 - yxu@suse.de
> 
> - include sassman's patch: fix-sched_yield_1_1.patch and
>   fix-sched_setparam_10_1.patch

Author: Stefan Assmann <sassmann@suse.de>

---

Index: ltp-full-20061017/testcases/open_posix_testsuite/conformance/interfaces/sched_setparam/10-1.c
===================================================================
--- ltp-full-20061017.orig/testcases/open_posix_testsuite/conformance/interfaces/sched_setparam/10-1.c
+++ ltp-full-20061017/testcases/open_posix_testsuite/conformance/interfaces/sched_setparam/10-1.c
@@ -86,7 +86,7 @@ void child_process(){
 
 	while(1) {
 		(*shmptr)++;
-		sched_yield();
+	sched_yield();
 	}
 }
 
@@ -127,7 +127,7 @@ int main(){
 	}
 	*shmptr = 0;
 
-	param.sched_priority = sched_get_priority_max(SCHED_FIFO);
+	param.sched_priority = sched_get_priority_min(SCHED_FIFO) + 2;
 
 	if(sched_setscheduler(getpid(), SCHED_FIFO, &param) != 0) {
 		if(errno == EPERM) {
